utils/spdx/src/main/kotlin/Utils.kt

```diff
@@ -49,7 +49,9 @@ val scanCodeLicenseTextDir by lazy {
     val pythonBinDir = listOf("bin", "Scripts")
     val scanCodeBaseDir = scanCodeExeDir?.takeUnless { it.name in pythonBinDir } ?: scanCodeExeDir?.parentFile
 
-    scanCodeBaseDir?.walkTopDown()?.find { it.isDirectory && it.endsWith("licensedcode/data/licenses") }
+    scanCodeBaseDir?.walkTopDown()?.find { it.isDirectory && it.endsWith("licensedcode/data/licenses") } ?:
+        // In cases where Python version management tools like Pyenv are used, the directory structure differs.
+        scanCodeBaseDir?.parentFile?.walkTopDown()?.find { it.isDirectory && it.endsWith("licensedcode/data/licenses") }
```

Comment:
As this would also search `scanCodeBaseDir` again, maybe even before other siblings are searched and thus wastes time to search something that has already been searched, I believe we should either

a. only search `scanCodeBaseDir?.parentFile`,
b. search `scanCodeBaseDir?.parentFile` as a fallback but exclude `scanCodeBaseDir` then,
c. look into a solution based on running the `scancode-license-data` command as mentioned [here](https://github.com/oss-review-toolkit/ort/issues/8147#issuecomment-1902743430).

Reply:
> c. look into a solution based on running the `scancode-license-data` command as mentioned [here](https://github.com/oss-review-toolkit/ort/issues/8147#issuecomment-1902743430).

I guess we can scratch this option due to https://github.com/aboutcode-org/scancode-toolkit/issues/4024.

Reply:
Ok, `scancode-license-data --path license-data` actually works to dump the data to the `license-data`. Theoretically, we could run this as part of the Docker build to put the data into a known location, and just add that known location to the paths to search here.

Reply:
And I'd propose to use `/opt/scancode-license-data` as that path.

Reply:
@sschuberth I understand the reasoning behind the proposal to use the ScanCode script `scancode-license-data` to dump the license texts to a known location. But on the other side, I think we should go for a **quick bug fix first**, knowing there are potentially OSS Disclosure Documents out there that don't list the license texts and instead show grey empty boxes, potentially not fulfilling license obligations and getting people into trouble. A solution involving the `scancode-license-data` script would possibly also require to integrate this in the `ort requirements` command, would create new risks/errors that need to be handled during execution and so on. Can't we say _"let's do this quick bug fix to remove the compliance risks as fast as possible"_, and then do a more extensive solution in a future refactoring?

Reply:
Adapted the heuristic search algorithm in a way that the `scanCodeBaseDir` is not searched twice.

Reply:
> I think we should go for a **quick bug fix first**

The bug report has been open for almost one year. I don't think that spending a few more hours on a cleaner fix makes a difference.

> A solution involving the `scancode-license-data` script would possibly also require to integrate this in the `ort requirements` command

I don't think so. As `ort requirements` just makes use of `scanCodeLicenseTextDir`, it would only be a matter of adding the hard-coded path as a fallback to the logic in there.

> would create new risks/errors that need to be handled during execution and so on.

I don't get what you mean. `scancode-license-data` would be executed at Docker build time, and `scanCodeLicenseTextDir` would be extended to also look in `/opt/scancode-license-data` as a fallback. That's it, if I'm not overlooking something.

> Can't we say _"let's do this quick bug fix to remove the compliance risks as fast as possible"_, and then do a more extensive solution in a future refactoring?

Again, what's your rationale why this is urgent *now* out of a sudden, after having the bug lingering around for one year?

Reply:
Thanks for pointing out the alternative solution about provisioning a directory /opt/scan-code-license-data at Docker build time. I guess this is the best alternative.